### PR TITLE
config-diff: added ability to specify opt config keys.

### LINKED ISSUE #1643: rose config-diff: support optional configurations
We should have options to include optional configurations in the diff.

### PR DESCRIPTION
**Pull request by oliver-sanders**
Closes #1643 

@benfitzpatrick Please Review
@matthewrmshin Please Review

### CODE REVIEW

**Comment by matthewrmshin** [comment]
Please add new tests to test battery.

**Comment by oliver-sanders** [comment]
@matthewrmshin Added tests.

File: lib/python/rose/config.py [hunk 1]
```diff
@@ -640,7 +644,11 @@ def load_with_opts(self, source, node=None, more_keys=None,
             opt_conf_file_name = os.path.join(
                 source_dir, OPT_CONFIG_DIR, opt_conf_file_name_base)
             try:
-                self.load(opt_conf_file_name, node)
+                if mark_opt_confs:
+                    self.load(opt_conf_file_name, node, default_comments=[
+                        " setting from opt config \"%s\"" % (key,)])
```
Comment (matthewrmshin) [inline lib/python/rose/config.py]: Should the default comment template be a constant of the class?

**Comment by matthewrmshin** [comment]
Logic looks OK 2. (I have raised a question, however.) Test OK.

**Comment by benfitzpatrick** [comment]
Could we have `opt-conf-key-1` and `opt-conf-key-2` so that we can apply optional configurations to either of the 2 configurations to compare?

**Comment by oliver-sanders** [comment]
@benfitzpatrick Changed from opt-conf-key to opt-conf-key-1, 2.

**Comment by benfitzpatrick** [comment]
OK, good for me.

**Pull request merged**